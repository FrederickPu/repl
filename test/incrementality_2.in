{"cmd": "import Lean\nopen Lean"}

{"cmd": "example : True := by\n  dbg_trace \"!1\"\n  dbg_trace \"!2\"", "env": 0}

{"cmd": "example : True := by\n  dbg_trace \"!1\"\n  sorry", "env": 0}

{"cmd": "example : True := by\n  dbg_trace \"!1\"\n  dbg_trace \"!2\"", "env": 0, "incr": 1}

{"cmd": "example : True := by\n  dbg_trace \"!1\"\n  dbg_trace \"!2\"", "env": 0, "incr": 2}
