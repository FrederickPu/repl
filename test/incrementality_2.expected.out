{"incr": 0, "env": 0}

!1
!2
{"messages":
 [{"severity": "error",
   "pos": {"line": 1, "column": 18},
   "endPos": {"line": 3, "column": 16},
   "data": "unsolved goals\n⊢ True"},
  {"severity": "warning",
   "pos": {"line": 1, "column": 0},
   "endPos": {"line": 1, "column": 7},
   "data": "declaration uses 'sorry'"}],
 "incr": 1,
 "env": 1}

{"sorries":
 [{"proofState": 0,
   "pos": {"line": 3, "column": 2},
   "goal": "⊢ True",
   "endPos": {"line": 3, "column": 7}}],
 "messages":
 [{"severity": "warning",
   "pos": {"line": 1, "column": 0},
   "endPos": {"line": 1, "column": 7},
   "data": "declaration uses 'sorry'"}],
 "incr": 2,
 "env": 2}

{"messages":
 [{"severity": "error",
   "pos": {"line": 1, "column": 18},
   "endPos": {"line": 3, "column": 16},
   "data": "unsolved goals\n⊢ True"},
  {"severity": "warning",
   "pos": {"line": 1, "column": 0},
   "endPos": {"line": 1, "column": 7},
   "data": "declaration uses 'sorry'"}],
 "incr": 3,
 "env": 3}

!2
{"messages":
 [{"severity": "error",
   "pos": {"line": 1, "column": 18},
   "endPos": {"line": 3, "column": 16},
   "data": "unsolved goals\n⊢ True"},
  {"severity": "warning",
   "pos": {"line": 1, "column": 0},
   "endPos": {"line": 1, "column": 7},
   "data": "declaration uses 'sorry'"}],
 "incr": 4,
 "env": 4}

